### DISCUSSION TITLE
How to detect quantized PNG?

### DISCUSSION THREAD

**Discussion by toaster** [libvips/libvips#2436] [Q&A] [answered]
Hi,

I’m using libvips to transform images (mostly resize).
Currently, quantized PNGs (256 colors) become 32-bit RGB pics during the process.
In the end a shrunk image takes up more space than the original.
Therefore, I want to apply quantization on export iff the input was quantized.

Question: How can I detect with libvips if a loaded PNG was quantized?

Thanks in advance!

**Comment by jcupitt** [comment, marked as answer]
Hi @toaster,

libvips 8.10 added a metadata item for this -- `palette-bit-depth`:

```
$ vips copy k2.jpg x.png[palette]
$ vipsheader -a x.png
x.png: 1450x2048 uchar, 3 bands, srgb, pngload
width: 1450
height: 2048
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 2.835
yres: 2.835
filename: x.png
vips-loader: pngload
palette-bit-depth: 8
```

For palette PNGs it's set to the bit depth of the palette index, so 8 or 16.

**Reply by toaster** [reply]
Cool, is there any way to get this information with 8.8? I’m currently stuck with it because of other deps that would require a large amount of work to update.

**Reply by toaster** [reply]
Hmm, looks like it’s already there:

```shell
/ # vipsheader --vips-version
libvips 8.8.4-Thu Dec  5 16:24:34 UTC 2019
/ # vipsheader -a /mnt/data/q.png 
/mnt/data/q.png: 200x100 uchar, 3 bands, srgb, pngload
width: 200
height: 100
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 11.811
yres: 11.811
filename: /mnt/data/q.png
vips-loader: pngload
png-comment-0-Comment: Created with GIMP
palette-bit-depth: 8
```

**Reply by jcupitt** [reply]
Ooop, you're right, it was added in 8.8.
